src/thunderbots/software/grsim_communication/grsim_command_primitive_visitor.cpp

```diff
@@ -205,7 +205,50 @@ void GrsimCommandPrimitiveVisitor::visit(const MoveSpinPrimitive &move_spin_prim
 
 void GrsimCommandPrimitiveVisitor::visit(const PivotPrimitive &pivot_primitive)
 {
-    // TODO: https://github.com/UBC-Thunderbots/Software/issues/94
+    // compute final position
+    Point final_robot_position(
+        pivot_primitive.getPivotPoint().x() +
+            (pivot_primitive.getPivotRadius() * pivot_primitive.getFinalAngle().cos()),
+        pivot_primitive.getPivotPoint().y() +
+            (pivot_primitive.getPivotRadius() * pivot_primitive.getFinalAngle().sin()));
+
+    // get current vector from pivot point to robot position
+    Vector unit_pivot_point_to_robot_pos =
+        (pivot_primitive.getPivotPoint() - robot.position()).norm();
+
+    // get a vector in the tangential direction
+    Vector tangential_dir_1(unit_pivot_point_to_robot_pos.y(),
+                            -unit_pivot_point_to_robot_pos.x());
+    Vector tangential_dir_2(-unit_pivot_point_to_robot_pos.y(),
+                            unit_pivot_point_to_robot_pos.x());
+
+    // get collinear point on orbit, between the robot and the pivot point, used
+    // to maintain orbit
+    Point collinear_point_on_orbit =
+        pivot_primitive.getPivotPoint() +
+        pivot_primitive.getPivotRadius() * -unit_pivot_point_to_robot_pos;
+
+    // select orbit direction based on the magnitude of the vector connecting the robots
+    // next position to the final position
+    Vector tangential_vector =
+        (((collinear_point_on_orbit + tangential_dir_1) - final_robot_position).len() <
+         ((collinear_point_on_orbit + tangential_dir_2) - final_robot_position).len())
```

Comment:
Could we give make the two things we're comparing here into descriptively named variables? Would make this conditional a little easier to read.

Reply:
fixed